src/audio/module_adapter/module/waves/waves.c

```diff
@@ -456,6 +458,50 @@ static int waves_effect_revision(struct processing_module *mod)
 	return 0;
 }
 
+/* cache config blob*/
+static int waves_effect_cache_config_blob(struct processing_module *mod, void *data, uint32_t size)
+{
+	struct comp_dev *dev = mod->dev;
+	struct module_data *codec = &mod->priv;
+	struct waves_codec_data *waves_codec = codec->private;
+	int ret = 0;
+
+	comp_dbg(dev, "waves_effect_cache_config_blob() start");
+
+	/* release old cached config blob*/
+	if (waves_codec->config_blob && size != waves_codec->config_blob_size) {
+		comp_info(dev, "waves_effect_cache_config_blob() release blob");
+		module_free_memory(mod, waves_codec->config_blob);
+		waves_codec->config_blob = NULL;
+		waves_codec->config_blob_size = 0;
+	}
+
+	if (!waves_codec->config_blob) {
+		waves_codec->config_blob = module_allocate_memory(mod, size, 16);
+		if (!waves_codec->config_blob) {
+			comp_err(dev,
+					"waves_effect_cache_config_blob() failed to allocate %d bytes for config blob",
+					size);
+			ret = -ENOMEM;
+		} else {
+			waves_codec->config_blob_size = size;
+			memset(waves_codec->config_blob, 0, size);
+			ret = memcpy_s(waves_codec->config_blob, waves_codec->config_blob_size,
+						data, size);
+			if (ret) {
+				comp_err(dev,
+						"waves_effect_cache_config_blob(): failed to copy config blob %d",
+						ret);
+				module_free_memory(mod, waves_codec->config_blob);
+				waves_codec->config_blob_size = 0;
+				return ret;
```

Comment:
replace line 497 with 
ret = -ENOMEM;

this can: 
1. keep same with if part.
2. one return in line 502 is enough for this case.